### ISSUE TITLE
[Feature Request] Table rendering in editor

### ISSUE THREAD

**Issue by nsrosenqvist** [UberWriter/uberwriter#88]
Currently tables aren't rendered and it'd be great if there was support for them, perhaps by autocompleting snippets as in #86 that also allows for new rows to be inserted when tabbing in the last column of the last row.

**Comment by somas95** [comment]
Do you mean in the "edit view", right? I'm not sure how easy it is to implement that in GtkTextView, but I'll look at it. This is something that I wanted in UberWriter too

**Comment by nsrosenqvist** [comment]
Yeah, autocompletion of column separators would be a nice start, but maybe something more advanced and user friendly is possible too? Mark Text basically adds a table like how a full featured rich text editor usually does, but that might clash with how UberWriter don't do wysiwyg, but maybe something simpler that helps with creating and adding columns can be done?

**Comment by somas95** [comment]
Some random and unrelated ideas: 


- The wysiwyg table 


- Auto completion (tab adds column, enter adds row) 


- Auto width of cells: if a cell is wider than the others in the same column uberwriter would add tabs to those other cells to have all the "|" aligned 


- Assistant to create X*Y tables / shortcut to insert a predefined 2*2 table which could be expanded with the idea n.2 


- Auto formatting when pasting from a csv file 


(aka you copy a table from LO Calc and then paste it in uberwriter, getting a fully formatted table) 




More ideas are welcome. From my point of view, all of them would play fine together except the wysiwyg table and the assistant, which would be unnecessary. 




Also, they're not especially difficult to implement. The fun part will be to write a REGEXP to detect if you're in a table or not (that's how we detect if we're on a formula or formatted text, btw. More difficult than it seems) 




On Fri, Nov 23, 2018 at 2:35 PM +0100, "Niklas Rosenqvist" <notifications@github.com> wrote:












Yeah, autocompletion of column separators would be a nice start, but maybe something more advanced and user friendly is possible too? Mark Text basically adds a table like how a full featured rich text editor usually does, but that might clash with how UberWriter don't do wysiwyg, but maybe something simpler that helps with creating and adding columns can be done?

**Comment by wolfv** [comment]
there is a sublime text plugin (written in Python) that does this quite well: https://github.com/vkocubinsky/SublimeTableEditor
